bigquery/job.go

```diff
@@ -399,8 +405,26 @@ type JobStatistics struct {
 	ScriptStatistics *ScriptStatistics
 
 	// ReservationUsage attributes slot consumption to reservations.
+	// This field reported misleading information and will no longer be populated.
 	ReservationUsage []*ReservationUsage
 
+	// ReservationId indicates the name of the primary reservation assigned to
+	// this job. Note that this could be different than reservations reported
+	// in the reservation usage field if parent reservations were used to
+	// execute this job.
+	ReservationId string
+
+	// Edition is the name of edition corresponding to the reservation for
+	// this job at the time of this update.
+	//
+	// Possible values:
+	//   "RESERVATION_EDITION_UNSPECIFIED" - Default value, which will be treated
+	// as ENTERPRISE.
+	//   "STANDARD" - Standard edition.
+	//   "ENTERPRISE" - Enterprise edition.
+	//   "ENTERPRISE_PLUS" - Enterprise Plus edition.
+	Edition string
```

Comment:
For those kind of fields, usually we created a custom enum like type to represent the available values, can you change that ? 
```suggestion
	Edition ReservationEdition
```

and add the new enum like type:
```
// ReservationEdition is used to specify the name of edition corresponding to the reservation.
type ReservationEdition string

var (

	// UnspecifiedReservationEdtion is the default value, which will be treated as EnterpriseReservationEdition
	UnspecifiedReservationEdition ReservationEdition = "RESERVATION_EDITION_UNSPECIFIED"
        
	// EnterpriseReservationEdition represents the Standard edition.
	StandardReservationEdition ReservationEdition = "STANDARD"
	
        // EnterpriseReservationEdition represents the Enterprise edition.
	EnterpriseReservationEdition ReservationEdition = "ENTERPRISE"
	
       	// EnterprisePlusReservationEdition represents the Enterprise Plus edition.
	EnterprisePlusReservationEdition ReservationEdition = "ENTERPRISE_PLUS"
)
```